test/notifications_test.js

```diff
@@ -81,29 +85,40 @@ describe('notifications', function() {
 
   serverManager.inject();
 
-  beforeEach(function*() {
-    mq.connect(mqSocket);
-    yield chakram.post(subscriptionsUrl, subscription);
-  });
+  it('should mark reminder as errored if no subscription exists', function*() {
+    let hasOneMessageBeenSent = false;
+    waitForMqMessage().then(() => {
+      hasOneMessageBeenSent = true;
+    });
 
-  afterEach(function() {
-    mq.disconnect(mqSocket);
+    yield chakram.post(remindersUrl, inputs[0]);
+    yield waitUntilReminderHasStatus('family_name', 1, 'error-no-subscription');
+    expect(hasOneMessageBeenSent).deep.equal(false);
   });
```

Comment:
Yes, I have 😄 . We won't have a false positive on this test. One issue, though, the error message is very generic, and cannot be customized. You get `Error: condition not satified after 4999ms`. `until-promise` is the only lib I found that handles condition that contain promises.

I checked their code. I believe I can make a PR to customize the timeout message.